plugin/src/test/java/io/jenkins/plugins/analysis/core/steps/WarningChecksPublisherITest.java

```diff
@@ -0,0 +1,104 @@
+import io.jenkins.plugins.checks.api.ChecksOutput;
+import io.jenkins.plugins.checks.api.ChecksOutput.ChecksOutputBuilder;
+import io.jenkins.plugins.checks.api.ChecksStatus;
+
+import static io.jenkins.plugins.analysis.core.assertions.Assertions.*;
+
+public class WarningChecksPublisherITest extends IntegrationTestWithJenkinsPerTest {
+    private static final String OLD_REPORT_FILE = "checkstyle.xml";
+    private static final String NEW_REPORT_FILE = "checkstyle1.xml";
+
+    /**
+     * Verifies that {@link WarningChecksPublisher} constructs the {@link ChecksDetails} correctly
+     * with only new issues.
+     */
+    @Test
+    public void shouldCreateChecksDetailsWithNewIssuesAsAnnotations() {
+        FreeStyleProject project = createFreeStyleProjectWithWorkspaceFiles(OLD_REPORT_FILE);
+        enableCheckStyleWarnings(project);
+
+        // trigger a build using the report file with 4 issues
+        Run<?, ?> run = buildSuccessfully(project);
+        assertThat(getAnalysisResult(run))
+                .hasTotalSize(4)
+                .hasNewSize(0);
+
+        // trigger a build using the report file with 2 new issues
+        copyMultipleFilesToWorkspaceWithSuffix(project, NEW_REPORT_FILE);
+        run = buildSuccessfully(project);
+        assertThat(getAnalysisResult(run))
+                .hasTotalSize(6)
+                .hasNewSize(2);
+
+        // extract details from result
+        WarningChecksPublisher publisher = new WarningChecksPublisher(getResultAction(run));
+        ChecksDetails details = publisher.extractChecksDetails();
+
+        // verify extracted details
+        ChecksDetails expectedDetails = createChecksDetailsBasedOnReportFile();
+        assertThat(details)
+                .usingRecursiveComparison()
+                .ignoringFields("detailsURL")
+                .isEqualTo(expectedDetails);
+    }
+
+    private ChecksDetails createChecksDetailsBasedOnReportFile() {
+        ChecksDetailsBuilder builder = new ChecksDetailsBuilder()
+                .withName("CheckStyle")
+                .withStatus(ChecksStatus.COMPLETED)
+                .withConclusion(ChecksConclusion.SUCCESS)
+                .withDetailsURL("http://localhost:39121/jenkins/job/test0/2/");
+
+
+        ChecksOutput output = new ChecksOutputBuilder()
+                .withTitle("CheckStyle Warnings")
+                .withSummary("## 6 issues in total:\n"
+                        + "- ### 2 new issues\n"
+                        + "- ### 4 outstanding Issues\n"
+                        + "- ### 2 delta issues\n"
+                        + "- ### 0 fixed issues")
+                .withText("## Total Issue Statistics:\n* Error: 6\n* High: 0\n* Normal: 0\n* Low: 0\n"
+                        + "## New Issue Statistics:\n* Error: 2\n* High: 0\n* Normal: 0\n* Low: 0\n"
+                        + "## Delta Issue Statistics:\n* Error: 2\n* High: 0\n* Normal: 0\n* Low: 0\n")
+                .addAnnotation(new ChecksAnnotationBuilder()
```

Comment:
Do you have an example output that shows how this is rendered?

Reply:
No for this specific report :)

I just copied the checkstyle.xml file from other test cases and constructed the annotation based on what the warnings publisher should do for it.

Reply:
Ok, then I have to wait until I see it in action on my machine. I just wondered if `- ### ` is valid in markdown since the hashes are used for headers.

Reply:
Is there any specific reason that you sometimes use `*` and sometimes `-` for bullet points?

Reply:
No, I just googled `- ###` because I want the bullet points with smaller text than `##`.

The markdown just don't have a unified standard, `- ###` is valid in GitHub but not valid in somewhere else (say the preview tool: https://markdownlivepreview.com/)

Turns out `* ###` is supported by GitHub as well, just tried it. I'll make them consistent.